internal/index/manager/manager.go

```diff
@@ -821,6 +829,27 @@ func (mgr *Manager) getIndexesCopy(start int) ([]*index.Reader, indexReleaser) {
 	return indexes, mgr.lock(indexes)
 }
 
+func (mgr *Manager) SetClientConfig(config ClientConfig) ClientConfig {
+	mgr.clientConfig = config
+	return mgr.ClientConfig()
+}
+
+func (mgr *Manager) ClientConfig() ClientConfig {
+	c := make(chan ClientConfig)
+	mgr.jobs <- func() {
+		locks := uint(0)
+		for _, n := range mgr.usedIndexes {
+			locks += n
+		}
+		c <- ClientConfig{
```

Comment:
Can't you return `mgr.clientConfig` directly?

Reply:
lol yes